src/System.Text.JsonLab.Dynamic/System/Text/Json/JsonDynamicObject.cs

```diff
@@ -20,6 +21,78 @@ private JsonDynamicObject(Dictionary<JsonProperty, JsonValue> properties)
             _properties = properties;
         }
 
+        private static Dictionary<Type, Dictionary<int, PropertyInfoLinkedList>> TypeCache = new Dictionary<Type, Dictionary<int, PropertyInfoLinkedList>>();
+
+        public static T Deserialize<T>(ReadOnlySpan<byte> utf8)
```

Comment:
Shouldn't the API tell the called how many bytes were consumed?

Reply:
Atm, the entire json payload is expected to be within the input span, so all of the data would be consumed, in which case, bytesConsumed isn't necessary.